### DISCUSSION TITLE
Django 5.1 register_converter warning

### DISCUSSION THREAD

**Discussion by nrlulz** [encode/django-rest-framework#9495] [Potential Issue] [answered]
After upgrading to Django 5.1 I am seeing this warning:

```
  /python/lib/python3.11/site-packages/rest_framework/urlpatterns.py:108: RemovedInDjango60Warning: Converter 'drf_format_suffix' is already registered. Support for overriding registered converters is deprecated and will be removed in Django 6.0.
    register_converter(suffix_converter, converter_name)
```

Relevant discussion on Django issue tracker: https://code.djangoproject.com/ticket/35090

Digging into it a little further, it looks like `DefaultRouter` calls `register_converter` when building urls. Our project has several `DefaultRouter`s, so this converter is registered several times, which in effect does nothing since it just replaces what's there with the same thing.

The fix could be as simple as throwing an `if converter_name not in get_converters():` check above the `register_converter` call in `rest_framework.urlpatterns.format_suffix_patterns`?

**Comment by ivorbosloper** [comment, marked as answer]
The suggested fix has just been merged to master https://github.com/encode/django-rest-framework/pull/9512 .

**Reply by nrlulz** [reply]
Thank you!
